### ISSUE TITLE
Have the types collection be unaffected by Sync viewport visibility 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#3006]
Have the visibility of the Types blender collection be unaffected by Sync viewport visibility during activate_drawing()

**Comment by Andrej730** [comment]
If I understand correctly the problem is that we use `bpy.ops.object.hide_view_clear()` during activating the drawing view which unhides all objects in the scene and can mess up visibility setting user had for Types collection.

https://user-images.githubusercontent.com/9417531/232438550-e2feb032-3202-4405-9203-63c4f7898204.mp4

**Comment by Gorgious56** [comment]
IMO if we're thinking "Blender's way" here it's more of a workflow problem rather than a feature issue. Blender users should never use the eye icon visibility to semi-permanently hide objects. You can either use the monitor icon visibility (shared across view layers) or the collection exclusion toggle (only for the current view layer).

![image](https://user-images.githubusercontent.com/25156105/232450093-5c32666a-bd3b-4c75-92bb-4ba0c49c0659.png)

But unfortunately this will be hard to understand for newcomers. Having 3 different ways to hide an object, 4 if we count the render visibility, 7 if we count the *collection* visibility settings is a steep realization to make. The fact that the monitor icon is hidden by default in the outliner doesn't help this to be recognized. The fact that an object *can* be linked in several different collections, either. :)

So the solution may be to selectively hide / unhide objects, but that opens the possibility of missing something or rule bloat if we want to preserve some other collection down the line.

TLDR : eye icon visibility should *not* be used by the user to store semi-permanent visibility states.

**Comment by Andrej730** [comment]
Didn't know this monitor icon is even existed, I thought Blender just store `object.hide_viewport` somewhere internally and we can't change it from UI 😅

But the fact that it's hidden means that either way most of the users will use this eye icon, so shouldn't we account for that?

**Comment by theoryshaw** [comment]
> If I understand correctly the problem is that we use bpy.ops.object.hide_view_clear() during activating the drawing view which unhides all objects in the scene and can mess up visibility setting user had for Types collection.

Yes!

The eyeball is so tempting! ;)
